### ISSUE TITLE
Feature request: Lag publications

### ISSUE THREAD

**Issue by nicolaslopezj** [MasterAM/meteor-lag-methods#1]
Is posible to lag publications?

**Comment by MasterAM** [comment]
I believe it is.
It crossed my mind, but since I tend to do every meaningful data change in a method, I decided to put it on hold at the moment.

The approach should be similar to the one taken by this package, only wrapping `Meteor.publish()`, `Meteor.server.publish_handlers` and `Meteor.server.universal_publish_handlers`.

This might be a bit more complex because of the reactive computations taking place.

Anyway, the next thing that I am going to do is write a small plugin for the `constellation:console` tool for this package and then I might have time to figure this out.

**Comment by nicolaslopezj** [comment]
I made some progress (requires `meteorhacks:unblock`)

``` js
Meteor.startup(function() {
  var rootUrl = process.env.ROOT_URL;
  if (rootUrl !== 'http://localhost:3000/') {
    return;
  } else {
    console.log('Delaying publications...');
  }

  _.keys(Meteor.server.publish_handlers).map(function(key) {
    var func = Meteor.server.publish_handlers[key];
    Meteor.server.publish_handlers[key] = function() {
      this.unblock();
      Meteor._sleepForMs(1000);
      return func.apply(this, arguments);
    };
  });

  Meteor.server.universal_publish_handlers = Meteor.server.universal_publish_handlers.map(function(func) {
    return function() {
      this.unblock();
      Meteor._sleepForMs(1000);
      return func.apply(this, arguments);
    };
  });
});
```

**Comment by MasterAM** [comment]
Nice. `meteorhacks:unblock` makes it easy to use a common base for both method and publication lagging. Pretty much all of the configs of the lag-methods package (default and custom delay, exclusion, unblocking and force blocking) can be specified for the publication lagging as well.

The approach would probably be to create a refactored base package with all of the common pieces, 2 packages for methods and publications, one package that wraps both and a constellation front-end (probably only a single one, for both utilities).

I will try to find time for it during the weekend.

**Comment by MasterAM** [comment]
Publication delay is now implemented (see https://github.com/MasterAM/meteor-lag-publications).
The constellation control panel is almost ready and will be released within the next couple of days.
